From e1bd1dc207dae92cdb5b424226f89d1b4e4bb182 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <bhutchings@solarflare.com>
Date: Thu, 14 Nov 2013 01:05:58 +0000
Subject: [PATCH v5 004/164] net_tstamp: Improve kernel-doc for struct hwtstamp_config

Git-commit: e1bd1dc207dae92cdb5b424226f89d1b4e4bb182
Patch-mainline: v3.14-rc1
Reference: fate#317388
Target: sle11-sp4

Fix the name of the rx_filter field.

Remove text about 32/64-bit compatibility; this works just the same as
for most socket ioctls and as the structure is not allowed to grow
there is no need to remind anyone how to maintain it.

Add explanation about drivers changing the filter mode.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 include/linux/net_tstamp.h |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

Index: linux-3.0-SLE11-SP4/include/linux/net_tstamp.h
===================================================================
--- linux-3.0-SLE11-SP4.orig/include/linux/net_tstamp.h
+++ linux-3.0-SLE11-SP4/include/linux/net_tstamp.h
@@ -30,13 +30,13 @@ enum {
  *
  * @flags:	no flags defined right now, must be zero
  * @tx_type:	one of HWTSTAMP_TX_*
- * @rx_type:	one of one of HWTSTAMP_FILTER_*
+ * @rx_filter:	one of HWTSTAMP_FILTER_*
  *
  * %SIOCSHWTSTAMP expects a &struct ifreq with a ifr_data pointer to
- * this structure. dev_ifsioc() in the kernel takes care of the
- * translation between 32 bit userspace and 64 bit kernel. The
- * structure is intentionally chosen so that it has the same layout on
- * 32 and 64 bit systems, don't break this!
+ * this structure.  If the driver or hardware does not support the
+ * requested @rx_filter value, the driver may use a more general
+ * filter mode.  In this case @rx_filter will indicate the actual mode
+ * on return.
  */
 struct hwtstamp_config {
 	int flags;
